src/services/recim/match.ts

```diff
@@ -37,35 +37,72 @@ type CreatedMatchStats = {
   MatchID: number;
   StatusID: number;
   Score: number;
-  Sportsmanship: number;
+  SportsmanshipScore: number;
 };
 
 type PatchMatchStats = {
   TeamID: number;
   StatusID?: number;
   Score?: number;
-  Sportsmanship?: number;
+  SportsmanshipScore?: number;
 };
 
 type PatchMatch = {
-  Time: string;
+  StartTime: string;
   SurfaceID: number;
   StatusID: number;
   TeamIDs?: Array<number>;
 };
 
-type CreatedAttendance = {
+type MatchAttendance = {
+  TeamID: number;
+  Attendance: Attendance;
+};
+
+type UploadMatchAttendance = {
+  TeamID: number;
+  Attendance: UploadAttendance[];
+};
+
+type UploadAttendance = {
+  Username: string;
+};
+
+type Surface = {
+  ID: number;
+  Name: string;
+  Description: string;
+};
+
+type UploadSurface = {
+  Name?: string;
+  Description?: string; // location/capacity/misc details
+};
+
+type Attendance = {
   ID: number;
   MatchID: number;
-  ParticipantUsername: string;
+  Username: string;
+  TeamID: number;
 };
 
 //Match Routes
 const createMatch = (newMatch: UploadMatch): Promise<CreatedMatch> =>
   http.post('recim/matches', newMatch);
 
-const createMatchAttendance = (username: string, matchID: number): Promise<CreatedAttendance> =>
-  http.post(`recim/matches/${matchID}/attendance`, username);
+const updateMatchAttendance = (
+  attendance: UploadMatchAttendance,
+  matchID: number,
+): Promise<Attendance[]> => http.put(`recim/matches/${matchID}/attendance`, attendance);
+
+const updateAttendance = (matchID: number, attendance: Attendance): Promise<Attendance> =>
+  http.post(`recim/matches/${matchID}/attendance`, attendance);
+
+const removeAttendance = (matchID: number, attendance: Attendance): Promise<Attendance> =>
+  http.del(`recim/matches/${matchID}/attendance`, attendance);
```

Comment:
Does this route actually need the entire `Attendance` object in the body? Typically, you just need the ID of the attendance to be deleted.